packages/react-button/.eslintrc.json

```diff
@@ -1,4 +1,12 @@
 {
   "extends": ["plugin:@fluentui/eslint-plugin/react"],
-  "root": true
+  "root": true,
+  "overrides": [
+    {
+      "files": ["*.stories.{ts,tsx}"],
```

Comment:
If we go forward with this change or similar, this rule should move to the overrides in `@fluentui/eslint-plugin/react`.

Also a more general comment, I have a hard time understanding the value of lint rules with "warning" level. If people decide to disregard the warning, it either causes spam in logs (annoying and potentially confusing) or requires adding a disable comment (same as would happen with an error level rule that someone felt should be ignored in a particular spot). However I'm definitely not blocking on this because based on common presets it's clear that a lot of the community doesn't share my opinion.

Reply:
> Note: This is just a showcase that package/domain specific configuration for linting works as expected

Got it, that makes sense.

I am actually interested in discussing general ideas of how linting ought to work some other time and considering whether our current approach is actually the right one (that was kind of why I mentioned the error thing), but agreed that it's not important to discuss now and not blocking this PR.